auth_interfaces.go

```diff
@@ -102,4 +105,10 @@ type RoleManager interface {
 	// RegisterPermissions adds a list of strings to the role manager as valid permission keys. Returns an
 	// error if the same permission is registered more than once
 	RegisterPermissions([]string) error
+
+	// FindRolesWithPermissions returns a role that exactly matches the given resources and permissions
+	FindRoleWithPermissions(string, []string, Permissions) (*Role, error)
```

Comment:
maybe `FindRoleWithResourcesAndPermissions` ?

Reply:
It seems too wordy to me. I think it's implied that you need to pass specific resources